edgelet/iotedged/src/lib.rs

```diff
@@ -1289,15 +1694,19 @@ mod tests {
         let crypto = TestCrypto {
             use_expired_ca: false,
             fail_device_ca_alias: true,
+            fail_decrypt: false,
+            fail_encrypt: false,
         };
         let mut tokio_runtime = tokio::runtime::Runtime::new().unwrap();
         let result = check_settings_state(
-            tmp_dir.path().to_path_buf(),
+            &tmp_dir.path().to_path_buf(),
```

Comment:
Once you've fixed `check_settings_state` to take `&Path` instead of `&PathBuf`, this can be `tmp_dir.path()`